File: backend/src/main/java/com/NBE_4_5_2/Team5/domain/post/category/service/CategoryService.java
package com.NBE_4_5_2.Team5.domain.post.category.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.NBE_4_5_2.Team5.domain.post.category.entity.Category;
import com.NBE_4_5_2.Team5.domain.post.category.repository.CategoryRepository;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class CategoryService {
	private final CategoryRepository categoryRepository;

	public List<Category> getCategories() {
		return categoryRepository.findAll();
	}
}

File: backend/src/main/java/com/NBE_4_5_2/Team5/domain/post/category/service/CategoryService.kt
package com.NBE_4_5_2.Team5.domain.post.category.service

import com.NBE_4_5_2.Team5.domain.post.category.entity.Category
import com.NBE_4_5_2.Team5.domain.post.category.repository.CategoryRepository
import org.springframework.stereotype.Service

@Service
class CategoryService(
    val categoryRepository: CategoryRepository
) {

    val categories: List<Category>
        get() = categoryRepository.findAll()
}
